test/integration/idle_timeout_integration_test.cc

```diff
@@ -181,5 +190,67 @@ TEST_P(IdleTimeoutIntegrationTest, PerStreamIdleTimeoutRequestAndResponse) {
   testRouterRequestAndResponseWithBody(1024, 1024, false, nullptr);
 }
 
+TEST_P(IdleTimeoutIntegrationTest, RequestTimeoutDisarmsOnRequestResponse) {
+  enable_request_timeout_ = true;
+  enable_per_stream_idle_timeout_ = false;
+
+  testRouterHeaderOnlyRequestAndResponse(true, nullptr);
+}
+
+TEST_P(IdleTimeoutIntegrationTest, RequestTimeoutDisarmsOnRequestResponseWithBody) {
+  enable_request_timeout_ = true;
+  enable_per_stream_idle_timeout_ = false;
+
+  testRouterRequestAndResponseWithBody(1024, 1024, false, nullptr);
+}
+
+TEST_P(IdleTimeoutIntegrationTest, RequestTimeoutTriggersOnBodilessPost) {
+  enable_request_timeout_ = true;
+
+  auto response = setupPerStreamIdleTimeoutTest("POST");
+
+  waitForTimeout(*response, "downstream_rq_path_timeout");
+
+  EXPECT_FALSE(upstream_request_->complete());
+  EXPECT_EQ(0U, upstream_request_->bodyLength());
+  EXPECT_TRUE(response->complete());
+  EXPECT_STREQ("408", response->headers().Status()->value().c_str());
+  EXPECT_EQ("request timeout", response->body());
+}
+
+TEST_P(IdleTimeoutIntegrationTest, UnconfiguredRequestTimeoutDoesNotTrigger) {
+  enable_per_stream_idle_timeout_ = true;
+  enable_request_timeout_ = false;
+
+  auto response = setupPerStreamIdleTimeoutTest("POST");
+
+  waitForTimeout(*response);
+
+  // with no request timeout configured, the idle timeout triggers instead
+  EXPECT_FALSE(upstream_request_->complete());
+  EXPECT_EQ(0U, upstream_request_->bodyLength());
+  EXPECT_TRUE(response->complete());
+  EXPECT_STREQ("408", response->headers().Status()->value().c_str());
+  EXPECT_NE("request timeout", response->body());
+}
+
+TEST_P(IdleTimeoutIntegrationTest, RequestTimeoutTriggersOnIncompleteHeaders) {
+  if (downstreamProtocol() == Envoy::Http::CodecClient::Type::HTTP2) {
+    return;
+  }
+
+  enable_request_timeout_ = true;
+
+  initialize();
+  fake_upstreams_[0]->set_allow_unexpected_disconnects(true);
+
+  std::string raw_response;
+  sendRawHttpAndWaitForResponse(lookupPort("http"), "GET / HTTP/1.1", &raw_response);
+  EXPECT_THAT(raw_response, testing::HasSubstr("request timeout"));
+}
+
+// TODO create a test filter that hangs and does not send data upstream, which would
```

Comment:
Do you mean to address this as part of this PR? Otherwise, it should be in the format `// TODO(auni53) ...`

Reply:
+1.  If you can, adding the test with this PR is best, since it's testing something that's part of the PR.